calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/metadata/DocumentDbTableSchemaGeneratorBasicTest.java

```diff
@@ -0,0 +1,1471 @@
+        Assertions.assertEquals(1, metadata.size());
+        Assertions.assertEquals(COLLECTION_NAME,
+                metadata.get(COLLECTION_NAME).getSqlName());
+        final DocumentDbMetadataTable metadataTable = (DocumentDbMetadataTable) metadata.get(COLLECTION_NAME);
+        Assertions.assertNotNull(metadataTable);
+        Assertions.assertEquals(13, metadataTable.getColumnMap().size());
+        final Set<JdbcType> integerSet = metadataTable.getColumnMap().values().stream().collect(
+                Collectors.groupingBy(DocumentDbSchemaColumn::getSqlType)).keySet();
+        Assertions.assertEquals(9, integerSet.size());
+        final Set<JdbcType> expectedTypes = ImmutableSet.of(
+                JdbcType.BIGINT,
+                JdbcType.VARBINARY,
+                JdbcType.BOOLEAN,
+                JdbcType.DECIMAL,
+                JdbcType.DOUBLE,
+                JdbcType.INTEGER,
+                JdbcType.NULL,
+                JdbcType.TIMESTAMP,
+                JdbcType.VARBINARY,
+                JdbcType.VARCHAR
+        );
+        Assertions.assertTrue(expectedTypes.containsAll(integerSet));
+
+        final Map<String, String> tableNameMap = new HashMap<>();
+        // Check for in-order list of columns.
+        int columnIndex = 0;
+        for (Entry<String, DocumentDbSchemaColumn> entry : metadataTable.getColumnMap()
+                .entrySet()) {
+            Assertions.assertTrue(0 != columnIndex
+                    || toName(combinePath(COLLECTION_NAME, "_id"), tableNameMap).equals(entry.getKey()));
+            Assertions.assertTrue(1 != columnIndex || "fieldDouble".equals(entry.getKey()));
+            Assertions.assertTrue(2 != columnIndex || "fieldString".equals(entry.getKey()));
+            Assertions.assertTrue(3 != columnIndex || "fieldObjectId".equals(entry.getKey()));
+            Assertions.assertTrue(4 != columnIndex || "fieldBoolean".equals(entry.getKey()));
+            Assertions.assertTrue(5 != columnIndex || "fieldDate".equals(entry.getKey()));
+            Assertions.assertTrue(6 != columnIndex || "fieldInt".equals(entry.getKey()));
+            Assertions.assertTrue(7 != columnIndex || "fieldLong".equals(entry.getKey()));
+            Assertions.assertTrue(8 != columnIndex || "fieldMaxKey".equals(entry.getKey()));
+            Assertions.assertTrue(9 != columnIndex || "fieldMinKey".equals(entry.getKey()));
+            Assertions.assertTrue(10 != columnIndex || "fieldNull".equals(entry.getKey()));
+            Assertions.assertTrue(11 != columnIndex || "fieldBinary".equals(entry.getKey()));
+            Assertions.assertTrue(12 != columnIndex || "fieldDecimal128".equals(entry.getKey()));
+            columnIndex++;
+        }
+
+        printMetadataOutput(metadata, getMethodName());
+    }
+
+    /**
+     * Tests a two-level document.
+     */
+    @DisplayName("Tests a two-level document.")
+    @Test
+    void testComplexTwoLevelDocument() {
```

Comment:
It can be confusing when there are some test with the word "complex" in their names inside "DocumentDbTableSchemaGenerator**Basic**Test". Maybe name

Reply:
Using Basic/Complex/Advanced can be very subjective.
I rather use other criteria to split the class, or add a comment into the class level explaining what is considered basic and what is complex

Reply:
Agree this is pretty random and confusing. 😅  Tried to split in a new way, ending up with 3 new files: 
- Identifier - checks identifier truncation logic for column and table names
- VirtualTable - more focused on testing if we get correct set of virtual tables and foreign keys from documents with nested structure
- Column - more focused on testing correct column type promotion